examples/repository/example_test.go

```diff
@@ -0,0 +1,206 @@
+
+		t.Run("ExampleGenericService_Truncate",
+			func(t *testing.T) {
+				t.Parallel()
+				ExampleGenericService_Truncate()
+			})
+
+		t.Run("ExampleGenericService_Upsert",
+			func(t *testing.T) {
+				t.Parallel()
+				ExampleGenericService_Upsert()
+			})
+
+		t.Run("ExampleGenericService_ListTables",
+			func(t *testing.T) {
+				t.Parallel()
+				ExampleGenericService_ListTables()
+			})
+
+		t.Run("ExampleGenericService_ListPrimaryKeys",
+			func(t *testing.T) {
+				t.Parallel()
+				ExampleGenericService_ListPrimaryKeys()
+			})
+	}
+}
+
+func ExampleNew() {
+	dsn := os.Getenv("DATABASE_URL")
+	ctx := context.TODO()
+
+	repo, err := repository.New(ctx, dsn)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(repo.Storage.IsNoSQL())
+	// Output:
+	// true
+}
+
+func ExampleNewTx() {
+	dsn := os.Getenv("DATABASE_URL")
+	ctx := context.TODO()
+
+	txRepo, err := repository.NewTx(ctx, dsn)
+	if err != nil {
+		panic(err)
+	}
+
+	req := &proto.UpsertRequest{
+		Table:    "TxnExampleTable",
+		Data:     []byte(`[{"id": "7fd0abc0-e5ad-4cbb-8d54-f2b3f43364da"}]`),
+		DataType: int32(tools.UpsertDataJSON),
+	}
+
+	rsp, err := txRepo.Upsert(ctx, req)
+	if err != nil {
+		panic(err)
+	}
+
+	if err := txRepo.Commit(); err != nil {
+		panic(err)
+	}
+
+	fmt.Println(rsp.GetUpsertedCount())
+	// Output:
+	// 0
+}
+
+func ExampleGenericService_Truncate() {
+	ctx := context.Background()
+	dns := os.Getenv("DATABASE_URL")
+
+	repo, err := repository.New(ctx, dns)
+	if err != nil {
+		panic(err)
+	}
+
+	req := &proto.TruncateRequest{
+		Tables: []string{"ExampleTable"},
+	}
+
+	rsp, err := repo.Truncate(ctx, req)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(rsp.DeletedCount)
+	// Output:
+	// 0
+}
+
+func ExampleGenericService_Upsert() {
+	ctx := context.Background()
+	dns := os.Getenv("DATABASE_URL")
+
+	repo, err := repository.New(ctx, dns)
+	if err != nil {
+		panic(err)
+	}
+
+	req := &proto.UpsertRequest{
+		Table:    "ExampleTable",
+		Data:     []byte(`[{"id": "7fd0abc0-e5ad-4cbb-8d54-f2b3f43364da"}]`),
+		DataType: int32(tools.UpsertDataJSON),
+	}
+
+	rsp, err := repo.Upsert(ctx, req)
+	if err != nil {
+		panic(err)
+	}
+
+	tally := rsp.GetMatchedCount()
+	fmt.Println(tally)
+	// Output:
+	// 0
+}
+
+func ExampleGenericService_ListTables() {
+	var err error
+
+	ctx := context.TODO()
+	dsn := os.Getenv("DATABASE_URL")
+
+	repo, err := repository.New(ctx, dsn)
+	if err != nil {
+		panic(err)
+	}
+
+	rsp, err := repo.ListTables(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(len(rsp.TableSet))
+	// Output:
+	// 2
+}
+
+func ExampleGenericService_ListPrimaryKeys() {
+	var err error
+
+	ctx := context.TODO()
+	dsn := os.Getenv("DATABASE_URL")
+
+	repo, err := repository.New(ctx, dsn)
+	if err != nil {
+		panic(err)
+	}
+
+	rsp, err := repo.ListPrimaryKeys(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	totalPKeys := 0
+	for _, keys := range rsp.PKSet {
+		totalPKeys += len(keys.GetList())
+	}
+
+	fmt.Println(totalPKeys)
+	// Output:
+	// 2
```

Comment:
```suggestion
```

Reply:
Sure thing, ill make the change right now then push.